internal/impl/postgresql/input_postgrecdc.go

```diff
@@ -0,0 +1,562 @@
+// the License. You may obtain a copy of the License at
+//
+// https://github.com/redpanda-data/connect/v4/blob/main/licenses/rcl.md
+
+package pgstream
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"strconv"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/Jeffail/checkpoint"
+	"github.com/jackc/pgx/v5/pgconn"
+	"github.com/lucasepe/codename"
+	"github.com/redpanda-data/benthos/v4/public/service"
+
+	"github.com/redpanda-data/connect/v4/internal/impl/postgresql/pglogicalstream"
+)
+
+var randomSlotName string
+
+type asyncMessage struct {
+	msg   service.MessageBatch
+	ackFn service.AckFunc
+}
+
+var pgStreamConfigSpec = service.NewConfigSpec().
+	Beta().
+	Categories("Services").
+	Version("0.0.1").
+	Summary(`Creates a PostgreSQL replication slot for Change Data Capture (CDC)
+		== Metadata
+
+This input adds the following metadata fields to each message:
+- streaming (Indicates whether the message is part of a streaming operation or snapshot processing)
+- table (Name of the table that the message originated from)
+- operation (Type of operation that generated the message, such as INSERT, UPDATE, or DELETE)
+		`).
+	Field(service.NewStringField("host").
+		Description("The hostname or IP address of the PostgreSQL instance.").
+		Example("123.0.0.1")).
+	Field(service.NewIntField("port").
+		Description("The port number on which the PostgreSQL instance is listening.").
+		Example(5432).
+		Default(5432)).
+	Field(service.NewStringField("user").
+		Description("Username of a user with replication permissions. For AWS RDS, this typically requires superuser privileges.").
+		Example("postgres"),
+	).
+	Field(service.NewStringField("password").
+		Description("Password for the specified PostgreSQL user.")).
+	Field(service.NewStringField("schema").
+		Description("The PostgreSQL schema from which to replicate data.")).
+	Field(service.NewStringField("database").
+		Description("The name of the PostgreSQL database to connect to.")).
+	Field(service.NewStringEnumField("tls", "require", "none").
+		Description("Specifies whether to use TLS for the database connection. Set to 'require' to enforce TLS, or 'none' to disable it.").
+		Example("none").
+		Default("none")).
+	Field(service.NewBoolField("stream_uncomited").
+		Description("If set to true, the plugin will stream uncommitted transactions before receiving a commit message from PostgreSQL. This may result in duplicate records if the connector is restarted.").
+		Default(false)).
+	Field(service.NewStringField("pg_conn_options").
+		Description("Additional PostgreSQL connection options as a string. Refer to PostgreSQL documentation for available options.").
+		Default(""),
+	).
+	Field(service.NewBoolField("stream_snapshot").
+		Description("When set to true, the plugin will first stream a snapshot of all existing data in the database before streaming changes.").
+		Example(true).
+		Default(false)).
```

Comment:
@Jeffail did we want to decouple the snapshot from the change stream as two different plugins?